### PR TITLE
430 - Low LHS on product pages. part 1: 395 - Dynamically load subsequent tabs on product & application pages

### LINKED ISSUE #395: Dynamically load subsequent tabs on product & application pages
For complex product and application pages we should try out `IntersectionObserver` to increase page loading speed and reduce initial DOM size.

- [ ] all tabs except "Overview" should be loaded only when becoming visible
- [ ] deep linking to tabs using #xxx must still work
- [ ] do not negatively effect other pages using tabs like /about-us

### LINKED ISSUE #430: Low LHS on product pages
Product pages with lots of resources give bad LHS because they have to load lots of resources. We need to improve this.

### PR DESCRIPTION
**Pull request by andreituicu**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Partial #430
Fix #395
Load eager only the visible tab and lazy load the rest.

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers
- After: https://430-products-lhs--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers

- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#applications
- After: https://430-products-lhs--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#applications

- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#resources
- After: https://430-products-lhs--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#resources

- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#citations
- After: https://430-products-lhs--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#citations

- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#order
- After: https://430-products-lhs--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#order

- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#related-products-services
- After: https://430-products-lhs--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers#related-products-services

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -223,6 +223,43 @@ function decoratePageNav(main) {
...
+          loadLazyBlocks(section);
+        }
+      });
+      observer.observe(section);
+
+      // Render the section with a delay
+      setTimeout(() => {
+        observer.disconnect();
+        loadLazyBlocks(section);
+      }, 3500);
```
Comment (jose-correia) [inline scripts/scripts.js:260]: I think we should extract this code into a separate function, with proper docstring with why its needed

**Review by jose-correia — CHANGES_REQUESTED** [review]

File: scripts/scripts.js [hunk 2]
```diff
@@ -196,6 +196,63 @@ function decorateLinks(main) {
...
+ * @param {string} nameOfFirstSection Exact name of the first section, in case there is no hash
+ */
+function lazyLoadHiddenPageNavTabs(sections, nameOfFirstSection) {
+  const activeHash = window.location.hash;
+  const active = activeHash ? activeHash.substring(1, activeHash.length) : nameOfFirstSection;
+
+  sections.forEach((section) => {
+    if (section.getAttribute('aria-labelledby') !== active) {
+      /*
+       1. It marks all the blocks inside the hidden sections as loaded,
```
Comment (duynguyen) [inline scripts/scripts.js:211]: LGTM.
minor: this step is numbered 1 but no 2, 3, 4, etc.

**Review by duynguyen — APPROVED** [review]

File: scripts/scripts.js (hunk 2 above)
Comment (andreituicu) [reply scripts/scripts.js]: Sorry, changed my mind :)

File: scripts/scripts.js (hunk 1 above)
Comment (andreituicu) [reply scripts/scripts.js]: Done. thanks! 👍

**Review by jose-correia — APPROVED** [review]

**Pull request merged**